components/brave_wallet/browser/account_discovery_manager.cc

```diff
@@ -163,4 +200,51 @@ void AccountDiscoveryManager::ProcessDiscoveryResult(
   }
 }
 
+void AccountDiscoveryManager::OnBitcoinDiscoverAccountsDone(
+    base::expected<DiscoveredBitcoinAccount, std::string> dicovered_account) {
+  if (!dicovered_account.has_value()) {
+    return;
+  }
+
+  auto& acc = dicovered_account.value();
+  if (acc.next_unused_receive_index == 0 && acc.next_unused_change_index == 0) {
+    return;
+  }
+
+  CHECK(IsBitcoinKeyring(acc.keyring_id));
+
+  mojom::AccountIdPtr last_bitcoin_account;
+  for (const auto& account : keyring_service_->GetAllAccountInfos()) {
+    const auto& account_id = account->account_id;
+    if (account_id->coin == mojom::CoinType::BTC &&
+        account_id->keyring_id == acc.keyring_id) {
+      if (!last_bitcoin_account || last_bitcoin_account->bitcoin_account_index <
+                                       account_id->bitcoin_account_index) {
+        last_bitcoin_account = account_id->Clone();
+      }
+    }
+  }
+
+  if (last_bitcoin_account &&
+      last_bitcoin_account->bitcoin_account_index + 1 != acc.account_index) {
+    return;
+  }
```

Comment:
A comment here about why we early return would be nice, I assume it's because we don't want to create accounts with gap?

Reply:
yep!
https://github.com/brave/brave-core/pull/20634/commits/14fd36bf729975cdbf628608b1de08abdeee5272